### ISSUE TITLE
Can't transpile `actix-web` VIA `cargo build --release --target=x86_64-unknown-linux-musl`

### ISSUE THREAD

**Issue by colbyn** [actix/actix-web#2329]
# New Bug ([GitHub repo](https://github.com/colbyn/actix-web-bug-2020-7-13))

Can't transpile `actix-web` VIA `cargo build --release --target=x86_64-unknown-linux-musl`
```
➜  actix-web-bug-2020-7-13 git:(main) ./build.sh
   Compiling ring v0.16.20
   Compiling brotli-sys v0.3.2
   Compiling unicode-normalization v0.1.19
   Compiling quote v1.0.9
   Compiling iovec v0.1.4
   Compiling net2 v0.2.37
   Compiling signal-hook-registry v1.4.0
   Compiling num_cpus v1.13.0
error: failed to run custom build command for `brotli-sys v0.3.2`

Caused by:
  process didn't exit successfully: `/Users/colbyn/Scratch/actix-web-bug-2020-7-13/target/release/build/brotli-sys-c7b71fef0074ab07/build-script-build` (exit code: 1)
  --- stdout
  cargo:include=/Users/colbyn/.cargo/registry/src/github.com-1ecc6299db9ec823/brotli-sys-0.3.2/brotli/include
  TARGET = Some("x86_64-unknown-linux-musl")
  OPT_LEVEL = Some("3")
  HOST = Some("x86_64-apple-darwin")
  CC_x86_64-unknown-linux-musl = None
  CC_x86_64_unknown_linux_musl = None
  TARGET_CC = None
  CC = None
  CROSS_COMPILE = None
  CFLAGS_x86_64-unknown-linux-musl = None
  CFLAGS_x86_64_unknown_linux_musl = None
  TARGET_CFLAGS = None
  CFLAGS = None
  CRATE_CC_NO_DEFAULTS = None
  DEBUG = Some("false")
  CARGO_CFG_TARGET_FEATURE = Some("fxsr,sse,sse2")
  running: "musl-gcc" "-O3" "-ffunction-sections" "-fdata-sections" "-fPIC" "-m64" "-I" "brotli/include" "-o" "/Users/colbyn/Scratch/actix-web-bug-2020-7-13/target/x86_64-unknown-linux-musl/release/build/brotli-sys-fbf7e23b3c7ec250/out/brotli/common/dictionary.o" "-c" "brotli/common/dictionary.c"

  --- stderr
  fatal: not a git repository (or any of the parent directories): .git


  error occurred: Failed to find tool. Is `musl-gcc` installed?


warning: build failed, waiting for other jobs to finish...
error: build failed
```

## Steps to Reproduce (for bugs)

Setup:
```
rustup target add x86_64-unknown-linux-musl
```

Run:
```
git clone https://github.com/colbyn/actix-web-bug-2020-7-13
cd actix-web-bug-2020-7-13
cargo build --release --target=x86_64-unknown-linux-musl
```

## Your Environment

- MacOS, stable.

- Rust Version
 ```
 $ rustc -V
 rustc 1.52.1 (9bc8c42bb 2021-05-09)
 ```
- Actix Web Version:
 ```
 [dependencies]
 actix-web = {version = "3.3.2", features = ['rustls']}
 ```

**Comment by robjtede** [comment]
musl is not officially supported, the logs seem to indicate that brotli crate is the culprit

**Comment by colbyn** [comment]
Okay, I'm still trying to figure out what happened since this used to work. For a second I though this was a new dependency from `actix-web` but the last `3.x` release was several months ago. Likewise `brotli-sys` hasn't been updated in a few years.  (Although I suppose it could be from the actual C dependency. If so, I'm surprised this worked in the past.)

**Comment by colbyn** [comment]
Okay, well I suppose I posted this in the wrong place.

Anyway, I don't know what caused this phenomena. But if others stumble into the same issue, here is what I did to get `actix-web` (or more specifically `brotli-sys`) cross compiling again. 

First of all, I forgot to add `.cargo/config` to the example git repo with the following:
```
[target.x86_64-unknown-linux-musl]
linker = "x86_64-linux-musl-gcc"
```

Apart from that, for some reason adding this fixes the issue at hand: 
```
export TARGET_CC=x86_64-linux-musl-gcc
```

Then calling
```
cargo build --release --target=x86_64-unknown-linux-musl
```
works!  🙂
